src/Avalonia.Controls/ItemsControl.cs

```diff
@@ -23,6 +23,7 @@ namespace Avalonia.Controls
     /// Displays a collection of items.
     /// </summary>
     [PseudoClasses(":empty", ":singleitem")]
+    [ItemType(typeof(ContentControl))]
     public class ItemsControl : TemplatedControl, IChildIndexProvider
```

Comment:
ItemsControl can have any types as items. ContentControl is just a default for virtualized containers.

Reply:
ItemType does not add any constraints, it is only used to allow Intellisense to propose the types.

I use this because it is default container that can be used to display an item.

https://github.com/AvaloniaUI/Avalonia/blob/8ea685eae40a9dd581bfd7ecf19f2f5604d21c17/src/Avalonia.Controls/ItemsControl.cs#L338-L344